akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -25,10 +25,19 @@ import scala.concurrent.duration.{ Duration, FiniteDuration }
  *  - otherwise, these streams do not interfere with the element flow, ordinary completion or failure
  */
 @InternalApi private[akka] object Timers {
-  private def idleTimeoutCheckInterval(timeout: FiniteDuration): FiniteDuration = {
+
+  /**
+   * Given a timeout computes how often the check should be run without causing
+   * excessive load or loosing timeout precision.
+   */
+  private[akka] def timeoutCheckInterval(timeout: FiniteDuration): FiniteDuration = {
```

Comment:
* added visibility modifier so it could be used in tests.
* renamed method because it's not used in idleTimeout-related stages (also `BackpressureTimeout`)

Reply:
👍

Reply:
IMHO this should have been documented as it impacts how precise the checks are; could you still add some docs maybe? At least on the class level or in the docs section that talks about the Timers? Expecting someone to be suprised about this at some point

Reply:
I more expect people to be surprised about the old behavior, setting a timeout to 100s and actually not getting the timeout until after 125s

Reply:
True, would be nice to document our logic for this anyway :~